### DISCUSSION TITLE
Automations: trigger when a row field is set to a specific value

### DISCUSSION THREAD

**Discussion by toomuchdesign** [Budibase/budibase#9499] [Help]
I'm trying to configure automation so that it triggers an action when a row is edited and a specific fields is set to a given value (eg: `status` -> `approved`).

I don't want the action to be triggered when the row is updated and the field had already the same value.

I tried to fetch the previous value/row but I couldn't.

Is there a way to configure automations task as described? Thank you all for the great work here!

**Comment by kasgithub8** [comment]
Hi @toomuchdesign,

I can think of two ways but if you want to trigger different automations for different field values, you could set the automation to "App Action":
![Screenshot 2023-02-01 150514](https://user-images.githubusercontent.com/95284278/216065129-3e63c937-8540-4ca4-b2b1-31bd71381c8e.png)

After that, use the button action "Trigger automation" and check "Use an existing automation":
![Screenshot 2023-02-01 150643](https://user-images.githubusercontent.com/95284278/216066093-0a728351-27ff-4933-a476-bdff9042a8e6.png)

**Reply by toomuchdesign** [reply]
Hi @kasgithub8,
thank you for the quick response, very appreciated!

The solution you describe involves triggering the automation "imperatively" through a UI interaction like a button click, right?

My initial idea was to bind the automation directly to the data "On row updated", so that any user interaction setting the mentioned field to the desired value triggers the action as a side-effect.

Do you think it would be possible?

**Reply by kasgithub8** [reply]
@toomuchdesign oh, my bad. The only thing I can think of is putting a condition on that field right after the automation start.
Is that something that could work?

**Reply by toomuchdesign** [reply]
I think it wouldn't work since you cannot get the previous row's values, therefore you cannot say whether a specific field was changed on the current update.

**Comment by Thornzee** [comment]
I am having the same issue! I have a form where I want to send an email to on person if the total amount is less than x but to another person if the amount is greater than x. I want to add a condition in my automation flow that evaluates the specific value in my trigger row.
